x-pack/plugins/enterprise_search/public/applications/analytics/components/analytics_overview/analytics_collections_logic.test.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter, mockFlashMessageHelpers } from '../../../__mocks__/kea_logic';
+
+import { nextTick } from '@kbn/test-jest-helpers';
+
+import { HttpError, Status } from '../../../../../common/types/api';
+
+import { AnalyticsCollectionsLogic } from './analytics_collections_logic';
+
+describe('analyticsCollectionsLogic', () => {
+  const { mount } = new LogicMounter(AnalyticsCollectionsLogic);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useRealTimers();
+    mount();
+  });
+
+  const DEFAULT_VALUES = {
+    analyticsCollections: [],
+    data: undefined,
+    hasNoAnalyticsCollections: false,
+    isLoading: true,
+    status: Status.IDLE,
+    isFirstRequest: true,
+  };
+
+  it('has expected default values', () => {
+    expect(AnalyticsCollectionsLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('reducers', () => {
+    describe('hasNoAnalyticsCollections', () => {
+      it('updates to true when apiSuccess returns empty analytics collections array', () => {
+        AnalyticsCollectionsLogic.actions.apiSuccess([]);
+        expect(AnalyticsCollectionsLogic.values.hasNoAnalyticsCollections).toBe(true);
+        expect(AnalyticsCollectionsLogic.values).toEqual(
+          expect.objectContaining({
```

Comment:
Small change for consistency: You already have some cases below, that checks all values at the same time.

In that case I think it would be something similar

```javascript
expect(AnalyticsCollectionsLogic.values.toEqual({
  ...DEFAULT_VALUES,
  // other fields that we are sure that changed.
  analyticsCollections: [],
  data: [],
  hasNoAnalyticsCollections: true,
  isFirstRequest: false,
  isLoading: false,
});
```

Reply:
Yeh - i'm lazy and used objectContaining. I think your approach is better as its only a couple more attributes that i needed to include to use without objectContaining. Updated!